.github/actions/build-setup/action.yml

```diff
@@ -0,0 +1,44 @@
+# SPDX-FileCopyrightText: Copyright 2025 Opal Health Informatics Group at the Research Institute of the McGill University Health Centre <john.kildea@mcgill.ca>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+name: Build App Setup
+description: Setup before building the app
+
+inputs:
+  NPMRC_FILE:
+    description: 'Secret with the contents of .npmrc'
+    required: true
+  ENV_CONFIG_JS:
+    description: 'Contents of the file `opal.config.js` for the given environment'
+    required: true
+  ENV_GOOGLE_SERVICES:
+    description: 'Contents of the file `google-services.json` for the given environment'
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    # Install dependencies
+    - name: Check npm installation
+      run: npm -v
+      shell: bash
+    - name: Load .npmrc
+      run: echo "${NPMRC_FILE}" > .npmrc
+      shell: bash
+      env:
+        NPMRC_FILE: ${{ inputs.NPMRC_FILE }}
+    - name: Install dependencies
+      run: npm ci --cache .npm --prefer-offline
```

Comment:
Do we cache `.npm` somewhere?

Reply:
Good point; I've updated the caching setup based on https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/caching-dependencies-to-speed-up-workflows